WcaOnRails/app/controllers/results_controller.rb

```diff
@@ -89,42 +122,55 @@ def rankings
               #{@region_condition}
               #{@gender_condition}
             ORDER BY value
-            #{limit_condition}
+            #{expanded_limit_condition}
           SQL
         end
         subquery = "(" + subqueries.join(") UNION ALL (") + ")"
         @query = <<-SQL
           SELECT *
-          FROM (#{subquery}) result
-          ORDER BY value, personName, competitionId, roundTypeId
-          #{limit_condition}
+          FROM (
+            SELECT
+              #{result_select},
+              RANK() OVER (ORDER BY result.value) rnk
+            FROM (#{subquery}) result
+            JOIN Competitions competition on competition.id = result.competitionId
+            ORDER BY value, competition.start_date, personName, competitionId, roundTypeId
+            #{expanded_limit_condition}
+          ) ranked
+          WHERE rnk <= #{@show}
         SQL
       end
     elsif @is_by_region
       @query = <<-SQL
-        SELECT
-          result.*,
-          result.#{value} value
+        SELECT *
         FROM (
           SELECT
-            result.countryId recordCountryId,
-            MIN(#{value}) recordValue
-          FROM Concise#{capitalized_type_param}Results result
+          #{result_select},
+          result.#{value} value,
+          RANK() over (order by result.#{value}) rnk
+          FROM (
+            SELECT
+              result.countryId recordCountryId,
+              MIN(#{value}) recordValue
+            FROM Concise#{capitalized_type_param}Results result
+            #{@gender_condition.present? ? "JOIN Persons persons ON result.personId = persons.id and persons.subId = 1" : ""}
+            WHERE 1
+              #{@event_condition}
+              #{@years_condition_result}
+              #{@gender_condition}
+            GROUP BY result.countryId
+          ) record
+          JOIN Results result ON result.#{value} = recordValue AND result.countryId = recordCountryId
+          JOIN Competitions competition on competition.id = competitionId
           #{@gender_condition.present? ? "JOIN Persons persons ON result.personId = persons.id and persons.subId = 1" : ""}
           WHERE 1
             #{@event_condition}
-            #{@years_condition_result}
```

Comment:
is this just formatting or did you really replace `years_condition_result` with` years_condition_competition` ? If so, why?

Reply:
This is just the formatting of the pr, the current code on master uses both in the same way